source/tests/test_file_generator/test_file_generator.cpp

```diff
@@ -0,0 +1,196 @@
+#include "test_file_generator.h"
+
+test_file_generator::test_file_generator()
+{
+    sodium_init();
+    make_file_name();
+    set_random();
+    add_claim(char_file_name);
+    read_claim(char_file_name);
+    make_buffer();
+    add_stack(char_file_name);
+};
+
+test_file_generator::test_file_generator(string test_case, uint64_t stack_number)
+{
+    sodium_init();
+    make_file_name();
+    if (test_case == "different block number")
+    {
+        set_random();
+        add_claim(char_file_name);
+        read_claim(char_file_name);
+        make_buffer();
+        change_numb(test_case, stack_number);
+        add_stack(char_file_name);
+    }
+    else if (test_case == "different user amount")
+    {
+        set_random();
+        add_claim(char_file_name);
+        read_claim(char_file_name);
+        make_buffer();
+        change_numb(test_case, stack_number);
+        add_stack(char_file_name);
+    }
+    else if (test_case == "broken payment id")
+    {
+        set_random();
+        add_claim(char_file_name);
+        read_claim(char_file_name);
+        make_buffer();
+        change_numb(test_case, stack_number);
+        add_stack(char_file_name);
+    }
+    else if(test_case == "broken start of payment id")
+    {
+        set_random();
+        change_numb(test_case, stack_number);
+        add_claim(char_file_name);
+        read_claim(char_file_name);
+        make_buffer();
+        add_stack(char_file_name);
+    }
+}
+
+test_file_generator::test_file_generator(string uuid)
+{
+    sodium_init();
+    make_file_name();
+    set_random();
+    add_claim(char_file_name);
+    read_claim(char_file_name);
+    make_buffer();
+    change_uuid(uuid);
+    add_stack(char_file_name);
+}
+
+void test_file_generator::change_uuid(string uuid)
+{
+    get_block_uuid = gen(uuid);
```

Comment:
It would be better, if `get_block_uuid` would be marked as **member instance**, we use the `m` prefix for this purposes. So the final name would be `mGetBlockUUID`.

But there is one more improvement here: `mGetBlockUUID` defines action (Get), but the variable can't be used in action, it only stores some data. Maybe it would be better to name it something like `mNextBlockUUID`?